src/psbt.h

```diff
@@ -668,6 +687,36 @@ struct PartiallySignedTransaction
                     }
                     break;
                 }
+                case PSBT_GLOBAL_XPUB:
+                {
+                    if (key.size() != BIP32_EXTKEY_WITH_VERSION_SIZE + 1) {
```

Comment:
23b45a66f3343fdf5315a8673ed898202fddf182: I assume that `+ 1` is here because `<xpub>` is encoded as CompactSize and is always shorter than 253 bytes  (i.e. it's always 78 bytes)? Still it seems cleaner to just deserialise it, rather than adding 1 everywhere.

Reply:
The `+ 1` is for the type which is always encoded as a single byte.

Reply:
I wish `<keylen>` would have been defined without the key type, but let's not overhaul the BIP :-)